File: app/android/app/src/main/java/com/jun/weather/ui/component/WeeklyItemView.java
package com.jun.weather.ui.component;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.constraintlayout.widget.ConstraintLayout;

import com.jun.weather.R;

public class WeeklyItemView extends ConstraintLayout {
    private TextView text_date;
    private TextView text_lowTemp;
    private TextView text_highTemp;
    private ImageView image_sky_day, image_sky_night;

    public WeeklyItemView(Context context) {
        super(context);
        initView(context);
    }

    public WeeklyItemView(Context context, AttributeSet attrs) {
        super(context, attrs);
        initView(context);
        getAttrs(attrs);
    }

    public WeeklyItemView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initView(context);
        getAttrs(attrs, defStyleAttr);
    }

    private void initView(Context context) {
        View view = LayoutInflater.from(context).inflate(R.layout.component_weekly_item, this, false);
        addView(view);

        text_date = findViewById(R.id.text_date);
        image_sky_day = findViewById(R.id.image_sky_day);
        image_sky_night = findViewById(R.id.image_sky_night);
        text_lowTemp = findViewById(R.id.text_lowTemp);
        text_highTemp = findViewById(R.id.text_highTemp);
    }

    private void getAttrs(AttributeSet attrs) {
        TypedArray typedArray = getContext().obtainStyledAttributes(attrs, R.styleable.WeeklyItemView);
        setTypeArray(typedArray);
    }

    private void getAttrs(AttributeSet attrs, int defStyleAttr) {
        TypedArray typedArray = getContext().obtainStyledAttributes(attrs, R.styleable.WeeklyItemView, defStyleAttr, 0);
        setTypeArray(typedArray);
    }

    private void setTypeArray(TypedArray typedArray) {
        text_date.setText(typedArray.getString(R.styleable.WeeklyItemView_date));
        image_sky_day.setImageResource(typedArray.getResourceId(R.styleable.WeeklyItemView_day_sky_image, 0));
        image_sky_night.setImageResource(typedArray.getResourceId(R.styleable.WeeklyItemView_night_sky_image, 0));
        text_lowTemp.setText(typedArray.getString(R.styleable.WeeklyItemView_low_temp));
        text_highTemp.setText(typedArray.getString(R.styleable.WeeklyItemView_high_temp));

        typedArray.recycle();
    }

    public void setDateText(String s) {
        text_date.setText(s);
    }

    public void setSkyDayImage(int resourceId) {
        image_sky_day.setImageResource(resourceId);
    }

    public void setSkyNightImage(int resourceId) {
        image_sky_night.setImageResource(resourceId);
    }

    public void setLowTemp(int i) {
        text_lowTemp.setText(String.valueOf(i));
    }

    public void setLowTemp(String val) {
        text_lowTemp.setText(val);
    }

    public void setHighTemp(int i) {
        text_highTemp.setText(String.valueOf(i));
    }

    public void setHighTemp(String val) {
        text_highTemp.setText(val);
    }
}

File: app/android/app/src/main/java/com/jun/weather/ui/component/WeeklyItemView.kt
package com.jun.weather.ui.component

import android.content.Context
import android.content.res.TypedArray
import android.os.Build
import android.text.Html
import android.text.SpannableString
import android.text.Spanned
import android.text.style.ForegroundColorSpan
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.ImageView
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.core.content.ContextCompat
import com.jun.weather.R
import com.jun.weather.databinding.ComponentWeeklyItemBinding
import java.util.*

class WeeklyItemView : ConstraintLayout {
    private lateinit var binding: ComponentWeeklyItemBinding

    constructor(context: Context) : super(context) {
        initView(context)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        initView(context)
        getAttrs(attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        initView(context)
        getAttrs(attrs, defStyleAttr)
    }

    private fun initView(context: Context) {
        val view = LayoutInflater.from(context).inflate(R.layout.component_weekly_item, this, false)
        addView(view)
        binding = ComponentWeeklyItemBinding.bind(view)
    }

    private fun getAttrs(attrs: AttributeSet?) {
        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.WeeklyItemView)
        setTypeArray(typedArray)
    }

    private fun getAttrs(attrs: AttributeSet?, defStyleAttr: Int) {
        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.WeeklyItemView, defStyleAttr, 0)
        setTypeArray(typedArray)
    }

    private fun setTypeArray(typedArray: TypedArray) {
        binding.textDate.text = typedArray.getString(R.styleable.WeeklyItemView_date)
        binding.imageSkyDay.setImageResource(typedArray.getResourceId(R.styleable.WeeklyItemView_day_sky_image, 0))
        binding.imageSkyNight.setImageResource(typedArray.getResourceId(R.styleable.WeeklyItemView_night_sky_image, 0))
//        binding.textLowTemp.text = typedArray.getString(R.styleable.WeeklyItemView_low_temp)
//        binding.textHighTemp.text = typedArray.getString(R.styleable.WeeklyItemView_high_temp)


        typedArray.recycle()
    }

    fun setDateText(s: String?) {
        binding.textDate.text = s
    }

    fun setSkyDayImage(resourceId: Int) {
        binding.imageSkyDay.setImageResource(resourceId)
    }

    fun setSkyNightImage(resourceId: Int) {
        binding.imageSkyNight.setImageResource(resourceId)
    }

    fun setTemp(low: Int, high: Int) {
        setTemp(low.toString(), high.toString())
    }

    fun setTemp(low: String, high: String) {
        val sp =  SpannableString(String.format(Locale.getDefault(), resources.getString(R.string.week_temp2), low,high))
        sp.setSpan(
                ForegroundColorSpan(ContextCompat.getColor(this.context, R.color.colorAccentBlue)),
                0,sp.indexOf("/"),Spanned.SPAN_INCLUSIVE_INCLUSIVE
        )
        sp.setSpan(
                ForegroundColorSpan(ContextCompat.getColor(this.context, R.color.colorAccentRed)),
                sp.indexOf("/"),sp.length,Spanned.SPAN_INCLUSIVE_INCLUSIVE
        )
    }
}